src/main/java/picard/vcf/GatherVcfs.java

```diff
@@ -61,32 +61,38 @@ public GatherVcfs() {
     protected int doWork() {
         log.info("Checking inputs.");
         INPUT = IOUtil.unrollFiles(INPUT, IOUtil.VCF_EXTENSIONS);
-        for (final File f: INPUT) IOUtil.assertFileIsReadable(f);
+        for (final File f : INPUT) IOUtil.assertFileIsReadable(f);
         IOUtil.assertFileIsWritable(OUTPUT);
 
-		final SAMSequenceDictionary sequenceDictionary = VCFFileReader.getSequenceDictionary(INPUT.get(0));
+        final SAMSequenceDictionary sequenceDictionary = VCFFileReader.getSequenceDictionary(INPUT.get(0));
 
         if (CREATE_INDEX && sequenceDictionary == null) {
             throw new PicardException("In order to index the resulting VCF input VCFs must contain ##contig lines.");
         }
 
         log.info("Checking file headers and first records to ensure compatibility.");
-        assertSameSamplesAndValidOrdering(INPUT);
-
-        if (areAllBlockCompressed(INPUT) && areAllBlockCompressed(CollectionUtil.makeList(OUTPUT))) {
-            log.info("Gathering by copying gzip blocks. Will not be able to validate position non-overlap of files.");
-            if (CREATE_INDEX) log.warn("Index creation not currently supported when gathering block compressed VCFs.");
-            gatherWithBlockCopying(INPUT, OUTPUT);
-        }
-        else {
-            log.info("Gathering by conventional means.");
-            gatherConventionally(sequenceDictionary, CREATE_INDEX, INPUT, OUTPUT);
+        try {
+            assertSameSamplesAndValidOrdering(INPUT);
+            if (areAllBlockCompressed(INPUT) && areAllBlockCompressed(CollectionUtil.makeList(OUTPUT))) {
+                log.info("Gathering by copying gzip blocks. Will not be able to validate position non-overlap of files.");
+                if (CREATE_INDEX)
+                    log.warn("Index creation not currently supported when gathering block compressed VCFs.");
+                gatherWithBlockCopying(INPUT, OUTPUT);
+            } else {
+                log.info("Gathering by conventional means.");
+                gatherConventionally(sequenceDictionary, CREATE_INDEX, INPUT, OUTPUT);
+            }
+        } catch (IllegalArgumentException e) {
```

Comment:
Why is important to catch this runtime exception when we throw another runtime exception, `PicardException`, above? I'm not familiar with how `doWork()` works; are some errors OK to catch and `return 1;` and others should crash the program?

Reply:
no...this is silliness on my part.

Reply:
in reading this again, the part that I care about is deleting what might be partially created vcf. given how the gathering works the truncated output might be valid...so it's important to delete it as the user might not be checking return codes.